pkg/ring/kv/memberlist/memberlist_client.go

```diff
@@ -673,64 +735,50 @@ func (m *Client) LocalState(join bool) []byte {
 // MergeRemoteState is method from Memberlist Delegate interface
 //
 // This is 'push' part of push/pull sync. We merge incoming KV store (all keys and values) with ours.
-func (m *Client) MergeRemoteState(stateMsg []byte, join bool) {
+func (m *KV) MergeRemoteState(data []byte, join bool) {
 	m.numberOfPushes.Inc()
-	m.totalSizeOfPushes.Add(float64(len(stateMsg)))
+	m.totalSizeOfPushes.Add(float64(len(data)))
 
-	buf := bytes.NewBuffer(stateMsg)
+	kvPair := KeyValuePair{}
 
 	var err error
-	for buf.Len() > 0 {
-		keyLen := byte(0)
-		keyLen, err = buf.ReadByte()
-		if err != nil {
+	for len(data) > 0 {
```

Comment:
I had a hard time understanding the slicing in this for loop. Could you add some comments explaining what `data` represents. It made more sense when I looked at some comments in other functions.

Reply:
Comment in `LocalState` was wrong (previous format), I've fixed it, and added comments into `MergeRemoteState` as well.